tests/std/tests/P0009R18_mdspan_mdspan/test.cpp

```diff
@@ -0,0 +1,1352 @@
+            requires is_constructible_v<extents_type, OtherExtents>
+        constexpr mapping(const mapping<OtherExtents>& other) : ActionTracker(other), mp(other.get_underlying()) {}
+
+        constexpr mapping& operator=(const mapping&) = default;
+        constexpr mapping& operator=(mapping&&)      = default;
+
+        constexpr const extents_type& extents() const {
+            return mp.extents();
+        }
+
+        template <class... IndexTypes>
+        constexpr index_type operator()(IndexTypes... indices) const {
+            return mp(indices...);
+        }
+
+        constexpr index_type required_span_size() const {
+            return mp.required_span_size();
+        }
+
+        constexpr bool is_unique() const {
+            return mp.is_unique();
+        }
+
+        constexpr bool is_exhaustive() const {
+            return mp.is_exhaustive();
+        }
+
+        constexpr bool is_strided() const {
+            return mp.is_strided();
+        }
+
+        constexpr index_type stride() const {
+            return mp.stride();
+        }
+
+        static constexpr bool is_always_unique() {
+            return underlying_mapping::is_always_unique();
+        }
+
+        static constexpr bool is_always_exhaustive() {
+            return underlying_mapping::is_always_exhaustive();
+        }
+
+        static constexpr bool is_always_strided() {
+            return underlying_mapping::is_always_strided();
+        }
+
+        constexpr bool operator==(const mapping& other) const {
+            return mp == other.mp;
+        }
+
+        constexpr bool is_constructed_with_extents_only() const noexcept
+            requires (!to_underlying(ReqId))
+        {
+            return constructed_with_extents_only;
+        }
+
+        constexpr const underlying_mapping& get_underlying() const noexcept {
+            return mp;
+        }
+
+        friend constexpr void swap(mapping& left, mapping& right) noexcept {
+            swap(static_cast<ActionTracker&>(left), static_cast<ActionTracker&>(right));
+            swap(left.mp, right.mp);
+        }
+
+    private:
+        bool constructed_with_extents_only = false;
+        underlying_mapping mp;
+    };
+};
+
+static_assert(check_layout_mapping_policy_requirements<TrackingLayout<layout_right>, dextents<int, 3>>());
+static_assert(check_layout_mapping_policy_requirements<TrackingLayout<layout_left>, dextents<int, 8>>());
+
+struct VectorBoolAccessor {
+    using offset_policy    = VectorBoolAccessor;
+    using element_type     = bool;
+    using reference        = vector<bool>::reference;
+    using data_handle_type = vector<bool>::iterator;
+
+    constexpr reference access(data_handle_type handle, size_t offset) const {
+        return handle[static_cast<ptrdiff_t>(offset)];
+    }
+
+    constexpr data_handle_type offset(data_handle_type handle, size_t offset) const {
+        return handle + static_cast<ptrdiff_t>(offset);
+    }

```

Comment:
It's slightly confusing to have a parameter `offset` shadow the member function `offset`. I'd recommend renaming the parameters to `off`. Occurs below.